js/script.js

```diff
@@ -0,0 +1,321 @@
+    document.getElementById("hideSource").addEventListener("click", HideSource);
+}, false);
+
+function CheckIsArray(array){
+    return Array.isArray(array);
+}
+
+function ShowCondition(conditionText){
+    let block = document.getElementById("task_condition");
+    if (block)
+    {
+        block.hidden = false;
+        let divs =block.getElementsByTagName("div");
+        if (divs && divs.length>0)
+        {
+            divs[0].innerHTML = conditionText;
+        }
+    }
+    document.getElementById("answer").hidden = true;
+    document.getElementById("additional").hidden = true;
+    document.getElementById("info").textContent = null;
+    document.getElementById("params").value = null;
+    document.getElementById("info2").hidden = true;
+    document.getElementById("info2").textContent = null;
+    document.getElementById("params2").hidden = true;
+    document.getElementById("params2").value = null;
+    clearError();
+    document.getElementById("result").removeEventListener("click", FuncToTask5);
+    document.getElementById("result").removeEventListener("click", FuncToTask6);
+    document.getElementById("result").removeEventListener("click", FuncToTask7);
+    document.getElementById("result").addEventListener("click", FuncToTask8);
+}
+
+function clearError(){
+    let error = document.getElementById("error");
+    error.textContent = null;
+    error.hidden = true;
+}
+
+function ShowAnswer(result){
+    let block = document.getElementById("answer");
+    if (block)
+    {
+        block.hidden = false;
+        let answerPlaces =block.getElementsByTagName("textarea");
+        if (answerPlaces && answerPlaces.length>0)
+        {
+            answerPlaces[0].textContent = JSON.stringify(result,null,'\t');
+        }
+    }
+}
+
+function ShowSource(){
+    let block = document.getElementById("source");
+    if (block)
+    {
+        block.hidden = false;
+        document.getElementById("showSource").hidden=true;
+        document.getElementById("hideSource").hidden=false;
+        let txtAreas =block.getElementsByTagName("textarea");
+        if (txtAreas && txtAreas.length>0)
+        {
+            txtAreas[0].textContent = JSON.stringify(sourceArray,null,'\t');
+        }
+    }
+}
+
+function HideSource(){
+    document.getElementById("hideSource").hidden=true;
+    document.getElementById("source").hidden=true;
+    document.getElementById("showSource").hidden=false;
+}
+
+function ShowAdditional(str){
+    let addBlock = document.getElementById("additional");
+    if (addBlock)
+    {
+        addBlock.hidden=false;
+        document.getElementById("info").textContent = str;
+    }
+}
+
+function ShowAdditional2(str){
+    let info = document.getElementById("info2");
+    let param = document.getElementById("params2");
+    if (info && param)
+    {
+        info.hidden=false;
+        param.hidden = false;
+        info.textContent = str;
+    }
+}
+
+function NewArrayWithDistinctElement(nameObject){
```

Comment:
https://www.syncfusion.com/blogs/post/10-javascript-naming-conventions-every-developer-should-know.aspx